internal/graph/reachableresources.go

```diff
@@ -36,11 +37,22 @@ type ValidatedReachableResourcesRequest struct {
 	Revision decimal.Decimal
 }
 
+type syncONRSet struct {
+	items sync.Map
+}
+
+func (s *syncONRSet) Add(onr *core.ObjectAndRelation) bool {
+	key := tuple.StringONR(onr)
+	_, existed := s.items.LoadOrStore(key, true)
```

Comment:
I think the empty struct instead of `true` trick to save memory should work for sync.Map, too.

Reply:
Looks like `nil` works too... maybe I should just use that?

Reply:
if the implementation of sync.Map uses a `map[any]any`, the go compiler will optimize `struct{}{}` to consume 0 bytes vs nil which is probably going to be sizeof(pointer)

Reply:
Updated